parser/shared/src/test/scala/hmda/parser/fi/FIGenerators.scala

```diff
@@ -12,9 +15,22 @@ trait FIGenerators {
     Gen.alphaStr
   }
 
-  // utility function(s)
+  // utility functions
 
   def stringOfN(n: Int, genOne: Gen[Char]): Gen[String] = {
     Gen.listOfN(n, genOne).map(_.mkString)
   }
+
+  // this name may be too similar to Gen.option. (to be fair, it's not unrelated. just... domain-specific.)
+  def optional[T](g: Gen[T], emptyVal: String = ""): Gen[String] = {
+    Gen.oneOf(g.map(_.toString), Gen.const(emptyVal))
+  }
+
+  def dateGen: Gen[String] = {
+    for {
+      year <- Gen.choose(2017, 2020) // or could parameterize if needed
+      month <- Gen.choose(1, 12)
+      day <- Gen.choose(1, 28) // NOTE: this misses some dates! if that's a problem, we can do something more complex.
+    } yield LocalDate.of(year, month, day).format(DateTimeFormatter.ofPattern("yyyyMMdd"))
```

Comment:
I suspect this is going to fail when trying to run it in Scala.js since it uses Java APIs that are not supported [1]. Have you tried this in a test? If not, could you write a small test that exercises this generator with dates to prove that it works in both environments? (something in the shared parser project should tell us what we need)

[1] http://www.lihaoyi.com/hands-on-scala-js/#JavaAPIs